iroh-resolver/src/balanced_tree.rs

```diff
@@ -33,7 +33,7 @@ impl TreeBuilder {
 
     pub fn stream_tree(
         &self,
-        chunks: impl Stream<Item = std::io::Result<BytesMut>>,
```

Comment:
Not sure how this is related to the PR, but I like it... It just does the .freeze() earlier it seems.

Reply:
it was, because I needed to restructure the stream output to abstract over the different chunkers